ttnn/cpp/ttnn/events.hpp

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <memory>
+
+#include "tt_metal/impl/device/device_mesh.hpp"
+
+namespace ttnn {
+
+namespace events {
+
+struct MultiDeviceEvent
+{
+    MultiDeviceEvent(DeviceMesh* device_mesh) {
+        auto& devices = device_mesh->mesh_devices;
+        this->events = std::vector<std::shared_ptr<Event>>(devices.size());
+        for (int event_idx = 0; event_idx < devices.size(); event_idx++) {
+            this->events[event_idx] = std::make_shared<Event>();
+            this->events[event_idx]->device = devices.at(event_idx).second;
+        }
+    }
+    std::vector<std::shared_ptr<Event>> events;
+};
+// Single Device APIs
+std::shared_ptr<Event> create_event(Device* device);
+void record_event(uint8_t cq_id, std::shared_ptr<Event> event);
+void wait_for_event(uint8_t cq_id, std::shared_ptr<Event> event);
+// Multi Device APIs
+MultiDeviceEvent create_event(DeviceMesh* device_mesh);
+void record_event(uint8_t cq_id, MultiDeviceEvent& event);
+void wait_for_event(uint8_t cq_id, MultiDeviceEvent& event);
+
+} // namespace events
+
+using namespace events;
```

Comment:
please don't write "using namespace" in hpp files.

Reply:
done